lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -2786,3 +2786,128 @@ def test_group_access(self, role_name, course_is_cohorted, thread_group_state):
             self.assertFalse(expected_error)
         except Http404:
             self.assertTrue(expected_error)
+
+
+@ddt.ddt
+class RetrieveThreadTest(
+        CommentsServiceMockMixin,
+        UrlResetMixin,
+        SharedModuleStoreTestCase
+):
+    """Tests for get_thread"""
+    @classmethod
+    @mock.patch.dict("django.conf.settings.FEATURES", {"ENABLE_DISCUSSION_SERVICE": True})
+    def setUpClass(cls):
+        super(RetrieveThreadTest, cls).setUpClass()
+        cls.course = CourseFactory.create()
+
+    @mock.patch.dict("django.conf.settings.FEATURES", {"ENABLE_DISCUSSION_SERVICE": True})
+    def setUp(self):
+        super(RetrieveThreadTest, self).setUp()
+        httpretty.reset()
+        httpretty.enable()
+        self.addCleanup(httpretty.disable)
+        self.thread_author = UserFactory.create()
+        self.register_get_user_response(self.thread_author)
```

Comment:
@bguertin 
This is being used for both 'text_basic' and 'test_thread_id_not_found' and any other test case if added in future could make use of it.
'test_nonauthor_enrolled_in_course' is a specific scenario that requires to create a new user and register its response.